Week-6/6.2/assignment6.2.js

```diff
@@ -0,0 +1,127 @@
+// constants for cursor traversal, we always start with Direction R
+const DIR_R = 'r';
+const DIR_D = 'd';
+const DIR_L = 'l';
+const DIR_U = 'u';
+
+function spiralIt(arr) {
+    // initialise new array for spiral element addition
+    let newArr = [];
+
+    const total_elements = arr.length * arr[0].length;
+
+    let dir = DIR_R;    
+
+    // initialise first limits
+    let rowStart = 0;
+    let rowEnd   = arr.length - 1;
+    let row = rowStart;
+
+    let colStart = 0;
+    let colEnd   = arr[0].length - 1;
+    let col = colStart;
+
+    // I tried with a while loop, but was getting undefined errors so just used a 'for'
+    for(let el=0 ; el<total_elements ; el++) {
```

Comment:
@mr-rego Looks good to me. Nice work. Please mention the time complexity in the comments.